CSharpBasics/CSharpAdvanced/Program.cs

```diff
@@ -0,0 +1,75 @@
+﻿#region Task_1
+
+using CSharpAdvanced;
+using CSharpOOP2;
+using System;
+
+int[] intArray = new int[] { 111, 25, -25, 56, 58, 777 };
+string[] stringArray = new string[] { "book", "door", "floor", "car" };
+decimal[] decimalArray = new decimal[] { 2.22m, 5.123m, 125.21m, 777.235m };
+
+Console.WriteLine(GetFirstElement(intArray));
+Console.WriteLine(GetFirstElement(stringArray));
+Console.WriteLine(GetFirstElement(decimalArray));
+
+static T GetFirstElement<T>(T[] item)
+{
+    return item.FirstOrDefault();
+}
+
+#endregion
+
+#region Task_2
+
+Chamomile chamomile = new Chamomile("Sun", "White", 0.25m);
+Rose rose = new Rose("Beatris", "Bordo", 1.11m);
+
+Console.Write("Enter number of roses : ");
+int rosesCount = int.Parse(Console.ReadLine());
+
+Console.Write("Enter number of chamomiles : ");
+int chamomilesCount = int.Parse(Console.ReadLine());
+
+Console.WriteLine($"\nFinal check for flowers: {Calculation.GetCheck(rose, rosesCount)}");
+
+Console.WriteLine($"\nFinal check for flowers: {Calculation.GetCheck(chamomile, chamomilesCount)}");
+
+#endregion
+
+#region Task_3
+
+var newChamomile = UpgradeFlowers<Flower>.RepaintFlower(chamomile);
```

Comment:
Try <Chamomile> :)
It is not crucial in this particular example, but it is better to use particular class :)